core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -156,7 +156,11 @@ private void attemptExecute(CompletableFuture<Response<OutputT>> future) {
             responseFuture.whenComplete((response, exception) -> {
 
                 if (exception != null) {
-                    maybeRetryExecute(future, exception);
+                    if (exception instanceof Error) {
+                        future.completeExceptionally(exception);
+                    } else {
+                        maybeRetryExecute(future, (Exception) exception);
```

Comment:
Is this cast necessary?

Reply:
Yes, exception is a Throwable, and new change to maybeRetryExecute() passes in an Exception instead of Throwable

Reply:
This assumes Throwable will always have only two top level interfaces: Error and Exception. Can we code defensively by doing instanceof check before casting?

Reply:
Updated